### PR TITLE
Finalizing project

### PR DESCRIPTION
**Pull request by Tommymarvel**
# Rails-Blog: Finalizing Project
## In this pull-request we did the followings:
- [x] Created a new rails project named `recipestash`.
- [x] Installed and configured `RSpec` & `Capybara` gems.
- [x] Setup the project **Kanban** board.
- [x] Added a login page.
- [x] Added a registration page.
- [x] Added a food list (with all CRUD implementation, except for 'update').
- [x] Added a recipes list (with all CRUD implementation, except for 'update').
- [x] Added a list of all public recipes from other users with their names and total prices.
- [x] Made sure that there are [no linter errors](https://github.com/microverseinc/linters-config).
- [x] Made sure that I used correct [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md).
- [x] Made sure that I documented your work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).
- [x] Made sure to follow the list of [best practices for Ruby](https://github.com/microverseinc/curriculum-ruby/blob/main/articles/ruby_best_practices.md).

### CODE REVIEW

File: app/views/foods/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+          <% @foods.each do | food | %>
+            <tr>
+              <td><%= food.name %></td>
+              <td><%= food.measurement_unit %></td>
+              <td>$<%= food.price %></td>
+              <td>
+                <%= link_to food, method: :delete, data: { turbo_method: :delete, turbo_confirm: 'Are you sure you want to delete this food?' }, class: "level-item" do %>
+                  <button class="button is-danger is-outlined is-rounded is-responsive">
+                    <span><ion-icon name="trash"></ion-icon></span>
+                  </button>
```
Comment (codepantha) [inline app/views/foods/index.html.erb]: - [x] According to requirements, only the owner of a food item can have the ability to delete it. Right now, in your app, anyone can delete any food item they choose. Please implement this requirement 👍

File: app/views/recipes/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+                                    <p><%= recipe.description %></p>
+                                </div>
+                                <nav class="level">
+                                    <div class="level-left is-mobile">
+                                        <%= link_to recipe, class: "level-item" do %>
+                                            <span class="icon"><i class="fa fa-link" aria-hidden="true"></i></span>
+                                        <% end %>
+                                        <%= link_to recipe, method: :delete, data: { turbo_method: :delete, turbo_confirm: 'Are you sure you wnat to delete this recipe?' }, class: "level-item" do %>
+                                            <span class="icon"><i class="fa fa-trash-o" aria-hidden="true"></i></span>
+                                        <% end %>
```
Comment (codepantha) [inline app/views/recipes/index.html.erb]: - [x] According to the project requirements, you can only remove a recipe item if it belongs to you. Right now, in your app, anyone can delete any recipe they choose even if it doesn't belong to them. Kindly implement this requirement 👍

File: app/views/recipes/_recipe.html.erb [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+          <div class="level-left is-mobile">
+              <%= link_to recipe, class: "level-item" do %>
+                <button class="button is-save is-outlined is-rounded is-responsive">Generate shopping list</button>
+              <% end %>
+          </div>
+          <div class="level-right is-mobile">
+              <!--%= link_to recipe_new_ingredient_path(recipe.id), class: "level-item", method: :get do %-->
+              <%= link_to recipe_new_ingredient_path(recipe.id), class: "level-item", method: :get do %>
+                <button class="button is-save is-outlined is-rounded is-responsive">Add ingredient</button>
+              <% end %>
```
Comment (codepantha) [inline app/views/recipes/_recipe.html.erb]: - [ ] The functionality to add ingredients to a recipe is not working as expected. Clicking on "Add Ingredient" simply reloads the page but the ingredient is never added to the recipe
image

File: app/views/recipes/_recipe.html.erb [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+            <%= recipe.cooking_time %>
+          </p>
+          <p><strong>Steps go here...</strong></p>
+        </div>
+        <br />
+        <nav class="level">
+          <div class="level-left is-mobile">
+              <%= link_to recipe, class: "level-item" do %>
+                <button class="button is-save is-outlined is-rounded is-responsive">Generate shopping list</button>
+              <% end %>
```
Comment (codepantha) [inline app/views/recipes/_recipe.html.erb]: - [ ] There seems to be a bug with the generate shopping list functionality as it doesn't work. Pressing the button to generate a shopping list simply does nothing. Please implement this functionality 🙏

File: app/views/recipes/index.html.erb [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+                        <article class="media">
+                            <div class="media-left">
+                                <figure class="image is-64x64">
+                                    <img src="https://via.placeholder.com/64.png">
+                                </figure>
+                            </div>
+                            <div class="media-content">
+                                <div class="content">
+                                    <strong><%= recipe.name %></strong><br />
+                                    <p><%= recipe.description %></p>
```
Comment (codepantha) [inline app/views/recipes/index.html.erb]: - [x] I noticed that you are listing all the recipes in the database, however, the recipes listed here should belong to the logged in user only. Kindly make this minor change 👍

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️ 

## Hi team 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights 🎯 
✔️ Descriptive PR
✔️ Used GitFlow correctly
✔️ Good commit history
✔️ No linter errors

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- n/a

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/recipes/_recipe.html.erb (hunk 4 above)
Comment (Tommymarvel) [reply app/views/recipes/_recipe.html.erb]: @codepantha this functionality is for 3 people group

**Comment by ZikaZaki** [comment]
Thanks @codepantha , we have already implemented all the requested changes. please give it another look.

File: app/views/foods/index.html.erb (hunk 1 above)
Comment (ZikaZaki) [reply app/views/foods/index.html.erb]: Thanks @codepantha for you time, we already implemented this rule. Please give it another look.

File: app/views/recipes/index.html.erb (hunk 2 above)
Comment (ZikaZaki) [reply app/views/recipes/index.html.erb]: Thanks @codepantha for you time, we already implemented this rule. Please give it another look.

File: app/views/recipes/index.html.erb (hunk 5 above)
Comment (ZikaZaki) [reply app/views/recipes/index.html.erb]: Thanks @codepantha for you time, we already implemented this rule. Please give it another look.

**Comment by ahmetbozaci** [comment]
### Hello @ZikaZaki @Tommymarvel 
### _This is an additional review as a result of the second opinion request._ 

- It appears you fix the all issues. 🥳 
___

 _Please note that since we will approve it manually so this could **take a few hours**_


### Cheers and Happy coding! 💻 👏
____
[GitHub](https://github.com/ahmetbozaci) | [LinkedIn](https://linkedin.com/in/ahmetbozaci) | [Twitter](https://twitter.com/ahmtbozaci)

_Reviewed by [Ahmet Bozacı ](https://github.com/ahmetbozaci)_

**Pull request merged**